src/pyird/utils/irdstream.py

```diff
@@ -151,21 +151,36 @@ def clean_pattern(self, trace_mask=None, hotpix_mask=None, extout='_cp', extin=N
             extin = self.extension
 
         extin_noexist, extout_noexist = self.check_existence(extin, extout)
-
+        # print('extin_noexist: ',extin_noexist)
+        # print('extout_noexist: ',extout_noexist)
         for i, fitsid in enumerate(tqdm.tqdm(extin_noexist)):
-            filen = self.rawdir/extin_noexist[i]
+            filen = self.rawdir / extin_noexist[i]
             hdu = pyf.open(filen)[0]
             im = hdu.data
             header = hdu.header
-            calim = np.copy(im)  # image for calibration
-            calim[trace_mask] = np.nan
-            if hotpix_mask is not None:
-                calim[hotpix_mask] = np.nan
-            model_im = median_XY_profile(calim)
-            corrected_im = im-model_im
+            if self.band == 'h':
+                calim = np.copy(im)  # image for calibration
+                calim[trace_mask] = np.nan
+                if hotpix_mask is not None:
+                    calim[hotpix_mask] = np.nan
+            elif self.band == 'y':
+                calim = np.copy(im)  # image for calibration
+                if hotpix_mask is not None:
+                    mask = trace_mask | hotpix_mask
+                else:
+                    mask = trace_mask
+                mask = mask.reshape(1, int(mask.size))
+                mask = mask[0][::-1].reshape(im.shape[0], im.shape[1]) # rotate mask matrix 180 degree
+                calim[mask] = np.nan
+
+            model_im = median_XY_profile(calim, show=False, Ncor=64, margin_npixel=4, sigma=0.2, xscale=32, yscale=64) # still need fine tuning
```

Comment:
I updated `median_XY_profile` in the precious PR, so could you change this to `model_im=median_XY_profile(calim)` to merge the current develop brunch?